tests/bgp/test_bgp_chasssis.py

```diff
@@ -0,0 +1,424 @@
+
+        routes = {}
+        routes[IPV4], routes[IPV6] = build_routes
+
+        def update_routes(exabgp_port, routes, asn=None, withdraw=False):
+            action = 'withdraw' if withdraw else 'announce'
+            for route in routes:
+                update_route(action, ptfhost.mgmt_ip, exabgp_port, route, asn)
+
+        def verify_routes(duthost, routes, neigh_ip_addrs, neigh_intfs,
+                          nexthops_with_inferior_paths, deleted):
+            nexthops = None if deleted else neigh_ip_addrs
+            num_internal_paths = 0
+            if neigh_intfs:
+                num_internal_paths = len(neigh_ip_addrs) - len(neigh_intfs)
+            else:
+                num_internal_paths = len(neigh_ip_addrs)
+
+            internal = bool(num_internal_paths)
+            for route in routes:
+                wait_until(30, 2, verify_bgp_route, duthost, route['prefix'], nexthops,
+                           nexthops_with_inferior_paths)
+                wait_until(30, 2, verify_ip_route, duthost, route['prefix'], nexthops)
+                if not deleted:
+                    verify_nexthop_interface(duthost, route['prefix'],
+                                             expected_external_intfs=neigh_intfs,
+                                             internal=internal,
+                                             num_internal_paths=num_internal_paths)
+
+        def get_neigh_info(dut, exabgp_ports, neigh_ip_addrs, neigh_intfs_by_dut,
+                           ipversion, neigh_index=0):
+            exabgp_port, neigh_ip_address, neigh_intf = \
+                                           get_exabpg_port(dut, ipversion, neigh_index)
+            exabgp_ports.append(exabgp_port)
+            neigh_ip_addrs.append(neigh_ip_address)
+            if not dut in neigh_intfs_by_dut:
+                neigh_intfs_by_dut[dut] = []
+
+            neigh_intfs_by_dut[dut].append(neigh_intf)
+
+        def update_routes_af(duthosts, ipversion, num_nexthops_per_dut, withdraw=False):
+            exabgp_ports = []
+            neigh_ip_addrs = []
+            neigh_intfs_by_dut = {}
+
+            for dut in duthosts:
+                num_nexthops = 1
+                if num_nexthops_per_dut and dut in num_nexthops_per_dut:
+                    num_nexthops = num_nexthops_per_dut[dut]
+                for neigh_index in range(num_nexthops):
+                    get_neigh_info(dut, exabgp_ports, neigh_ip_addrs,
+                                   neigh_intfs_by_dut, ipversion, neigh_index)
+
+            for exabgp_port in exabgp_ports:
+                update_routes(exabgp_port, routes[ipversion], withdraw=withdraw)
+
+            return (neigh_ip_addrs, neigh_intfs_by_dut)
+
+        def verify_routes_af(duthosts, allduts, ipversion,
+                             neigh_ip_addrs, neigh_intfs_by_dut,
+                             deleted=False):
```

Comment:
Please add comments just you did below. In not only helps in reviewing, will help later when enhancing or debugging this code.